src/opm/output/eclipse/AggregateWellData.cpp

```diff
@@ -0,0 +1,773 @@
+/*
+  Copyright 2018 Statoil ASA
+
+  This file is part of the Open Porous Media project (OPM).
+
+  OPM is free software: you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation, either version 3 of the License, or
+  (at your option) any later version.
+
+  OPM is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with OPM.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <opm/output/eclipse/AggregateWellData.hpp>
+
+#include <opm/output/data/Wells.hpp>
+
+#include <opm/output/eclipse/SummaryState.hpp>
+
+#include <opm/parser/eclipse/EclipseState/EclipseState.hpp>
+#include <opm/parser/eclipse/EclipseState/Runspec.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Schedule.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/ScheduleEnums.hpp>
+#include <opm/parser/eclipse/EclipseState/Schedule/Well.hpp>
+
+#include <opm/parser/eclipse/Units/UnitSystem.hpp>
+
+#include <algorithm>
+#include <cassert>
+#include <cstddef>
+#include <cstring>
+#include <iterator>
+#include <string>
+
+// #####################################################################
+// Class Opm::RestartIO::Helpers::AggregateWellData
+// ---------------------------------------------------------------------
+
+namespace {
+    std::size_t numWells(const std::vector<int>& inteHead)
+    {
+        // INTEHEAD(17) = NWELLS
+        return inteHead[17 - 1];
+    }
+
+    int maxNumGroups(const std::vector<int>& inteHead)
+    {
+        // INTEHEAD(21) = NWGMAX
+        return inteHead[21 - 1];
+    }
+
+    std::string trim(const std::string& s)
+    {
+        const auto b = s.find_first_not_of(" \t");
+        if (b == std::string::npos) {
+            // No blanks.  Return unchanged.
+            return s;
+        }
+
+        auto e = s.find_last_not_of(" \t");
+        if (e == std::string::npos) {
+            // No trailing blanks.  Return [b, end)
+            return s.substr(b, std::string::npos);
+        }
+
+        // Remove leading/trailing blanks.
+        return s.substr(b, e - b + 1);
+    }
+
+    std::vector<std::string>
+    groupNames(const std::vector<const Opm::Group*>& groups)
+    {
+        auto gnms = std::vector<std::string>{};
```

Comment:
You do like `auto` :-)

Reply:
It's object oriented :smirk:  Keeps the objects closer to the left margin.